[HZUNITY-19]: Extend SDK API around GDPR Consent Data

joaocunhajeronimo opened the pull request:
- New APIs for setting and clearing GDPR data implemented on the `HeyzapAds.cs` class.
- bridge between `C#` and Native code updated to support new GDPR Data APIs
  -  Android bridge is located in the SDK, see [android SDK changes](https://github.com/Heyzap/android-sdk/pull/615). In this repo those changes live within the `heyzap-ads-sdk.jar`
  - iOS bridge is located in the SDK, see [iOS SDK changes](https://github.com/Heyzap/ios-sdk/pull/752). In this repo this changes are duplicated in the `HeyzapUnityAds.mm` class
- bringing latest iOS SDK changes (`HeyzapAds.h` and respective lib `libHeyzapAds.a`) to support GDPR Data.
- bringing latest Android SDK changes (`heyzap-ads-sdk.jar`) to support GDPR Data
- changed the `TestScene.unity` so it's possible to test GDPR Data APIs from the test app. Respective scripts updated within `AdManager.cs`
  - updated `Readme.md` (and respective image) to reflect the new UI

The relevant Unity plugin API changes are replicated within the unity plugin repo.
see [related pull request](https://github.com/Heyzap/mobile-sdk/pull/145).

joaocunhajeronimo committed:
Adding support fro GDPR Data.
- New APIs for setting and clearing GDPR data implemented on the HeyzapAds.cs class.
- bridge between C# and Objectice-C code within HeyzapUnityAds.mm updated to support new GDPR Data APIs (Android bridge lives within the sdk so no changes required here)
- bringing latest iOS SDK changes (HeyzapAds.h and respective lib libHeyzapAds.a) to support GDPR Data.
- bringing latest Android SDK changes (heyzap-ads-sdk.jar) to support GDPR Data
- changed the TestScene.unity so it's possible to test GDPR Data APIs from the test app. Respective scripts updated within AdManager.cs
-- updated Reame.md (and respective image) to reflect the new UI

jjnunogarcia commented on Assets/Plugins/iOS/Heyzap/HeyzapAds.h:
```diff
@@ -325,6 +325,22 @@ extern NSString * const HZFyberVirtualCurrencyResponseUserInfoKey;
+ *
+ *  @param gdprConsentData The consent data
+ *
+ *  @since v9.21.0
```
Shouldn't it be `since 9.22.0`? I mean, the next version that will be released, no?

joaocunhajeronimo replied:
I think it doesn't matter for now. Anyways, I got the build from [proj-gdpr](https://github.com/Heyzap/ios-sdk/tree/proj-gdpr) which still defines 9.21.0 as its version

jjnunogarcia commented on Assets/Scripts/AdManager.cs:
```diff
@@ -265,6 +267,49 @@ public void RejectGdprButton()
+        HeyzapAds.SetGdprConsentData(gdprConsentData);
+    }
+
+    private Dictionary<string, string> getGdprConsentDataA()
```
I think the convention says to start with uppercase, if I'm not wrong?

joaocunhajeronimo replied:
true 👍

jjnunogarcia commented on Assets/Scripts/AdManager.cs:
```diff
@@ -265,6 +267,49 @@ public void RejectGdprButton()
+        return data;
+    }
+
+    private Dictionary<string, string> getGdprConsentDataB()
```
Same.

joaocunhajeronimo replied:
👍

jjnunogarcia commented:
As a general comment, there's no need to update HeyzapAds class, the jar file, etc. because it belongs to the mobile-sdk repo and it's there to be updated.
After releasing, we get the generated unity package and we import it into the test app. Then, those changes are merged.
But don't worry too much, it also doesn't hurt having them here, so I leave it up to you.

joaocunhajeronimo commented:
> After releasing, we get the generated unity package and we import it into the test app. Then, those changes are merged.

That might be the case... but the problem is that if I "request you to pull" you should be able to have a working version of the testapp.

joaocunhajeronimo committed:
Following C#'s upper camel case method naming convention on overlooked methods.
Fixing warning log message in case the clear GDPR consent data API is called outside of the mobile environment.

jjnunogarcia reviewed (approved):
Hmm I don't know if we have any kind of convention or process around that, but I guess what you say makes the review/test easier, so fine with me.
Regarding the PR, all good!

joaocunhajeronimo committed:
Reverted experimental integration code on AdManager. Changed log logic so it can also log null values.
Changed SetGdprConsntData implementation so that null values are supported. Chagned iOS bridge accordingly. (Android doesn't require changes).

joaocunhajeronimo committed:
typo
